modules/frontend/tag_handlers.go

```diff
@@ -221,6 +257,13 @@ func newTagsHTTPHandler(cfg Config, next pipeline.AsyncRoundTripper[combiner.Pip
 			return errResp, nil
 		}
 
+		if dataAccessController != nil {
+			if err := dataAccessController.HandleHTTPUnsupportedEndpoint(req); err != nil {
+				level.Error(logger).Log("msg", "SearchTags http: access control handling failed", "err", err)
+				return httpInvalidRequest(err), nil
```

Comment:
When `HandleHTTPUnsupportedEndpoint()` is called and it returns a `errUnsupportedEndpoint` this can be interpreted as accessing the endpoint is forbidden (status code 403), but `httpInvalidRequest()` maps the error to 400. Not 100% sure about this though and I guess this can be handled in a later PR.